tools/c7n_azure/c7n_azure/resources/storage.py

```diff
@@ -246,33 +244,102 @@ def process_resource_set(self, resources, event=None, session=None, token=None):
         return matched
 
     def _get_settings(self, storage_account, session=None, token=None):
-        if self.storage_type == self.BLOB_TYPE:
-            blob_property = get_annotation_prefix(self.BLOB_TYPE)
-            if not (blob_property in storage_account):
-                storage_account[blob_property] = json.loads(jsonpickle.encode(
-                    StorageSettingsUtilities.get_blob_settings(storage_account, token)))
-            return storage_account[blob_property]
-
-        elif self.storage_type == self.FILE_TYPE:
-            file_property = get_annotation_prefix(self.FILE_TYPE)
-            if not (file_property in storage_account):
-                storage_account[file_property] = json.loads(jsonpickle.encode(
-                    StorageSettingsUtilities.get_file_settings(storage_account, session)))
-            return storage_account[file_property]
-
-        elif self.storage_type == self.TABLE_TYPE:
-            table_property = get_annotation_prefix(self.TABLE_TYPE)
-            if not (table_property in storage_account):
-                storage_account[table_property] = json.loads(jsonpickle.encode(
-                    StorageSettingsUtilities.get_table_settings(storage_account, session)))
-            return storage_account[table_property]
-
-        elif self.storage_type == self.QUEUE_TYPE:
-            queue_property = get_annotation_prefix(self.QUEUE_TYPE)
-            if not (queue_property in storage_account):
-                storage_account[queue_property] = json.loads(jsonpickle.encode(
-                    StorageSettingsUtilities.get_queue_settings(storage_account, token)))
-            return storage_account[queue_property]
+        storage_prefix_property = ''
+        if self.storage_type == BLOB_TYPE:
+            storage_prefix_property = get_annotation_prefix(BLOB_TYPE)
+
+        elif self.storage_type == FILE_TYPE:
+            storage_prefix_property = get_annotation_prefix(FILE_TYPE)
+
+        elif self.storage_type == TABLE_TYPE:
+            storage_prefix_property = get_annotation_prefix(TABLE_TYPE)
+
+        elif self.storage_type == QUEUE_TYPE:
+            storage_prefix_property = get_annotation_prefix(QUEUE_TYPE)
+
+        if not (storage_prefix_property in storage_account):
+            settings = StorageSettingsUtilities.get_settings(
+                self.storage_type, storage_account, session, token)
+            storage_account[storage_prefix_property] = json.loads(jsonpickle.encode(settings))
+
+        return storage_account[storage_prefix_property]
+
+
+@Storage.action_registry.register('update-log-settings')
```

Comment:
nit we've been trying to move name idempotent modifiers with `set-` prefix instead of update/modify.